packages/spacecat-shared-aa-api-client/src/helpers.js

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { gzipSync } from 'zlib';
+/* eslint-env mocha */
+
+export const EVARS_KEY = 'eVars';
+export const EVENTS_KEY = 'events';
+
+export function getNextDay(dateString, offset) {
+  if (offset < 0) {
+    throw new Error('Offset must be a positive number');
+  }
+  const currentDate = !dateString ? new Date() : new Date(dateString);
+  currentDate.setUTCDate(currentDate.getDate() + (offset || 1));
+  return currentDate.toISOString().split('T')[0];
+}
+
+export function getPreviousDay(dateString, offset) {
+  if (offset < 0) {
+    throw new Error('Offset must be a positive number');
+  }
+  const currentDate = !dateString ? new Date() : new Date(dateString);
+  currentDate.setUTCDate(currentDate.getDate() - (offset || 1));
+  return currentDate.toISOString().split('T')[0];
+}
+
+export function parseDates(inputText) {
+  if (!inputText) {
+    return [];
+  }
+
+  const segments = inputText.split(';');
+
+  const dates = segments.reduce((acc, segment) => {
+    const trimmedSegment = segment.trim();
+    if (trimmedSegment.includes(',')) {
+      const [startDate, endDate] = trimmedSegment.split(',');
+      if (startDate > endDate) {
+        throw new Error('Start date cannot be after end date');
+      }
+      let currentDate;
+      try {
+        currentDate = new Date(`${startDate}T00:00:00.000Z`);
+      } catch (error) {
+        throw new Error(`Invalid start date ${startDate}`);
+      }
+      let finalDate;
+      try {
+        finalDate = new Date(`${endDate}T00:00:00.000Z`);
+      } catch (error) {
+        throw new Error(`Invalid end date ${endDate}`);
+      }
+      while (currentDate <= finalDate) {
+        acc.push(currentDate.toISOString().split('T')[0]);
+        currentDate.setUTCDate(currentDate.getDate() + 1);
+      }
+    } else if (trimmedSegment) {
+      acc.push(trimmedSegment);
+    }
+    return acc;
+  }, []);
+
+  return dates;
+}
+
+export function validateConfig(config) {
```

Comment:
I think validateConfig config should be added to spacecat-shared-data-access config model and renamed to validateRUMtoAAConfig